bookkeeper-server-shaded report NoClassDefFoundError

**BUG REPORT**

***Describe the bug***
 I am developing a flink state changelog storage  using bookkeeper-server-shaded. I get following error message. It seems that some of the dependencies are not being included in the shaded jar.
```
java.util.concurrent.ExecutionException: Boxed Error
        at scala.concurrent.impl.Promise$.resolver(Promise.scala:83)
        at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:75)
        at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:280)
        at akka.pattern.PromiseActorRef.$bang(AskSupport.scala:621)
        at akka.remote.DefaultMessageDispatcher.dispatch(Endpoint.scala:118)
        at akka.remote.EndpointReader$$anonfun$receive$2.applyOrElse(Endpoint.scala:1144)
        at akka.actor.Actor.aroundReceive(Actor.scala:537)
        at akka.actor.Actor.aroundReceive$(Actor.scala:535)
        at akka.remote.EndpointActor.aroundReceive(Endpoint.scala:540)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:580)
        at akka.actor.ActorCell.invoke(ActorCell.scala:548)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
        at akka.dispatch.Mailbox.run(Mailbox.scala:231)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1067)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1703)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:172)
Caused by: java.lang.NoClassDefFoundError: org/apache/bookkeeper/shaded/com/google/common/util/concurrent/internal/InternalFutureFailureAccess
        at org.apache.bookkeeper.shaded.com.google.common.util.concurrent.AbstractListeningExecutorService.newTaskFor(AbstractListeningExecutorService.java:45)
        at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:111)
        at org.apache.bookkeeper.shaded.com.google.common.util.concurrent.AbstractListeningExecutorService.submit(AbstractListeningExecutorService.java:56)
        at org.apache.bookkeeper.shaded.com.google.common.util.concurrent.ForwardingListeningExecutorService.submit(ForwardingListeningExecutorService.java:47)
        at org.apache.bookkeeper.common.util.BoundedScheduledExecutorService.submit(BoundedScheduledExecutorService.java:95)
        at org.apache.bookkeeper.shaded.com.google.common.util.concurrent.ForwardingListeningExecutorService.submit(ForwardingListeningExecutorService.java:47)
        at org.apache.bookkeeper.common.util.OrderedScheduler$OrderedSchedulerDecoratedThread.submit(OrderedScheduler.java:344)
        at org.apache.bookkeeper.common.util.OrderedScheduler$OrderedSchedulerDecoratedThread.submit(OrderedScheduler.java:302)
        at org.apache.bookkeeper.common.util.OrderedExecutor.<init>(OrderedExecutor.java:403)
        at org.apache.bookkeeper.common.util.OrderedScheduler.<init>(OrderedScheduler.java:116)
        at org.apache.bookkeeper.common.util.OrderedScheduler.<init>(OrderedScheduler.java:58)
        at org.apache.bookkeeper.common.util.OrderedScheduler$SchedulerBuilder.build(OrderedScheduler.java:78)
        at org.apache.bookkeeper.common.util.OrderedScheduler$SchedulerBuilder.build(OrderedScheduler.java:72)
        at org.apache.bookkeeper.client.BookKeeper.<init>(BookKeeper.java:429)
        at org.apache.bookkeeper.client.BookKeeper$Builder.build(BookKeeper.java:306)
        at org.apache.flink.changelog.bk.BkStateChangelogStorage.createBookKeeper(BkStateChangelogStorage.java:79)
        at org.apache.flink.changelog.bk.BkStateChangelogStorage.<init>(BkStateChangelogStorage.java:71)
        at org.apache.flink.changelog.bk.BkStateChangelogStorageFactory.createStorage(BkStateChangelogStorageFactory.java:50)
        at org.apache.flink.runtime.state.changelog.StateChangelogStorageLoader.load(StateChangelogStorageLoader.java:107)
        at org.apache.flink.runtime.state.TaskExecutorStateChangelogStoragesManager.stateChangelogStorageForJob(TaskExecutorStateChangelogStoragesManager.java:107)
        at org.apache.flink.runtime.taskexecutor.TaskExecutor.submitTask(TaskExecutor.java:702)
        at sun.reflect.GeneratedMethodAccessor12.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.lambda$handleRpcInvocation$1(AkkaRpcActor.java:309)
        at org.apache.flink.runtime.concurrent.akka.ClassLoadingUtils.runWithContextClassLoader(ClassLoadingUtils.java:83)
        at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleRpcInvocation(AkkaRpcActor.java:307)
        at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleRpcMessage(AkkaRpcActor.java:222)
        at org.apache.flink.runtime.rpc.akka.AkkaRpcActor.handleMessage(AkkaRpcActor.java:168)
        at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:24)
        at akka.japi.pf.UnitCaseStatement.apply(CaseStatements.scala:20)
        at scala.PartialFunction.applyOrElse(PartialFunction.scala:123)
        at scala.PartialFunction.applyOrElse$(PartialFunction.scala:122)
        at akka.japi.pf.UnitCaseStatement.applyOrElse(CaseStatements.scala:20)
        at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:171)
        at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:172)
        at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:172)
        at akka.actor.Actor.aroundReceive(Actor.scala:537)
        at akka.actor.Actor.aroundReceive$(Actor.scala:535)
        at akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:220)
        ... 9 more
```

--- a/shaded/bookkeeper-server-shaded/pom.xml
+++ b/shaded/bookkeeper-server-shaded/pom.xml
@@ -64,6 +64,7 @@
               <artifactSet>
                 <includes>
                   <include>com.google.guava:guava</include>
+                  <include>com.google.guava:failureaccess</include>
                   <include>com.google.protobuf:protobuf-java</include>
                   <include>org.apache.bookkeeper:bookkeeper-common</include>
                   <include>org.apache.bookkeeper:bookkeeper-common-allocator</include>

